src/multipart.lua

```diff
@@ -38,46 +38,79 @@ local function decode(body, boundary)
   part_value = {}
   part_value_ct = 0
 
-  for line in body:gmatch("[^\r\n]+") do
-    if line:sub(1, 2) == "--" and line:sub(3, #boundary+2) == boundary then
-      if part_name ~= nil then
-        result.data[part_index] = {
-          name = part_name,
-          headers = part_headers,
-          value = table.concat(part_value, "\r\n")
-        }
-
-        result.indexes[part_name] = part_index
-
-        -- Reset fields for the next part
-        part_headers = {}
-        part_value = {}
-        part_value_ct = 0
-        part_name = nil
-        part_index = part_index + 1
+  local position = 1
+  local done = false
+
+  repeat
+
+    local s = string.find(body, "[\r\n]", position)
+
+    local line
+
+    if s then
+      line = string.sub(body, position, s - 1)
+      position = s + 1
+    else
+      if position == 1 then
+        line = body
+      else
+        line = string.sub(body, position)
       end
-    elseif line:sub(1, 19):lower() == "content-disposition" then --Beginning of part
-      -- Extract part_name
-      for v in line:gmatch("[^;]+") do
-        if not is_header(v) then -- If it's not content disposition part
-          local pos = v:match("^%s*[Nn][Aa][Mm][Ee]=()")
-          if pos then
-            local current_value = v:match("^%s*([^=]*)", pos):gsub("%s*$", "")
-            part_name = string.sub(current_value, 2, string.len(current_value) - 1)
+
+      done = true
+    end
+
+    if line ~= "" then
```

Comment:
I'm not sure if I'm following this correctly, but what happens if you have a blank line split by `\r\n` inside one of the body parts? (ie, a blank line inside the "part value", which should be consumed in line 106)

Or is this not allowed by the multipart encoding standard, and if I send a DOS-formatted file I need to use Base64?